src/sstruct_mv/sstruct_matmult.c

```diff
@@ -217,79 +319,110 @@ hypre_SStructPMatmultSetup( hypre_SStructPMatmultData  *pmmdata,
 
    /* Create part grid data structure */
    hypre_SStructPGridCreate(comm, ndim, &pgrid);
-   hypre_SStructPGridSetVariables(pgrid, nvars, vartp);
+   hypre_SStructPGridSetVariables(pgrid, nvars, vartypes);
+   /* RDF: Need to figure out how to handle the cell grid (see 'RDF' below also) */
+   hypre_StructGridDestroy(hypre_SStructPGridCellSGrid(pgrid));
+   hypre_SStructPGridCellSGrid(pgrid) = NULL;
 
    /* Create part matrix data structure */
    hypre_SStructPMatrixCreate(comm, pgrid, pstencils, &pM);
    smaps = hypre_SStructPMatrixSMaps(pM);
 
+   /* Initialize the struct matmults for this part */
+   /* RDF NOTE: This does not assemble the struct grids.  They are assembled
+    * below or in HYPRE_SStructGridAssemble() to reduce box manager overhead.
+    * It's not clear if this is useful anymore. */
```

Comment:
This should still be useful to reduce communication overhead

Reply:
What I am referring to here is the `assemble_grid` feature of the `MatmultInit()` function.  I think we achieve the same communication reduction by way of the restructuring of the struct matmult interface, but I need to look more carefully (hence the RDF NOTE; you'll see that I left a number of these lying around :) ).

Reply:
Oh I see now, we could compare before vs after with caliper. I guess we need a couple hundreds of ranks at least to see a difference

Reply:
I haven't removed the feature yet, but caliper tests would still be good to do as part of our performance testing.  For now, I've just updated this comment to make it more clear.  Thanks!